typescript/phoenix-sdk/src/client.ts

```diff
@@ -804,6 +793,457 @@ export class Client {
   ): number {
     const market = this.markets.get(marketAddress);
     if (!market) throw new Error("Market not found: " + marketAddress);
-    return quoteAtoms / 10 ** market.quoteToken.data.decimals;
+    return market.quoteAtomsToQuoteUnits(quoteAtoms);
+  }
+
+  /**
+   * Instruction builders
+   **/
+
+  /**
+   * Creates a _CancelAllOrders_ instruction.
+   *
+   * @param marketAddress Market address string
+   * @param trader Trader public key (defaults to client's wallet public key)
+   *
+   * @category Instructions
+   */
+  public createCancelAllOrdersInstruction(
+    marketAddress: string,
+    trader?: PublicKey
+  ): TransactionInstruction {
+    if (!trader) {
+      trader = this.trader.pubkey;
+    }
+    const market = this.markets.get(marketAddress);
+    if (!market) throw new Error("Market not found: " + marketAddress);
+    return market.createCancelAllOrdersInstruction(trader);
```

Comment:
does `this.trader` always exist?

Reply:
It doesn't, can add an extra check if necessary or just punt